### PR TITLE
🐛 Destination BigQuery: Added more logs and updated the JobId made filled with locations and projectID

### LINKED ISSUE #3549: Destination Bigquery returns 404 when uploading data + resumable
## Expected Behavior
Sync between Shopify and BQ works

## Current Behavior
It's not totally clear why this happens... 
BigQuery returns 404 when uploading data after running a few hours.

From Google Cloud API Page [here](https://cloud.google.com/bigquery/docs/reference/api-uploads#best-practices)
> Handle 404 Not Found and 410 Gone errors when doing resumable uploads by starting the entire upload over from the beginning.

## Logs
```
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - Exception in thread "main" com.google.cloud.bigquery.BigQueryException: 404 Not Found
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - PUT https://www.googleapis.com/upload/bigquery/v2/projects/earthbones-shopify/jobs?uploadType=resumable&upload_id=ABg5-UyhG8f29jrwEfJJ2mu4JTAIjUF_B3F5JvymoctXdtBQjQ29LsavMFtadgBiXThGjQcTtZAffIZJTf0940HdexnlUOtGew
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.write(HttpBigQueryRpc.java:705)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.bigquery.TableDataWriteChannel$1.call(TableDataWriteChannel.java:59)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.bigquery.TableDataWriteChannel$1.call(TableDataWriteChannel.java:54)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.bigquery.TableDataWriteChannel.flushBuffer(TableDataWriteChannel.java:53)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.BaseWriteChannel.flush(BaseWriteChannel.java:112)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at com.google.cloud.BaseWriteChannel.write(BaseWriteChannel.java:139)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at io.airbyte.integrations.destination.bigquery.BigQueryDestination$RecordConsumer.acceptTracked(BigQueryDestination.java:358)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at io.airbyte.integrations.base.FailureTrackingAirbyteMessageConsumer.accept(FailureTrackingAirbyteMessageConsumer.java:66)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at io.airbyte.integrations.base.IntegrationRunner.consumeWriteStream(IntegrationRunner.java:132)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:113)
2021-05-21 03:10:39 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - 	at io.airbyte.integrations.destination.bigquery.BigQueryDestination.main(BigQueryDestination.java:421)
```

## Steps to Reproduce
It's not clear how to reproduce. Maybe this happens when trying to sync large Shopify datasets to BQ.

## Severity of the bug for you
Very low / Low / **Medium** / High / Critical

## Airbyte Version
0.23.0-alpha

## Connector Version (if applicable)
0.2.2 Shopify

## Additional context
Slack [convo](https://airbytehq.slack.com/archives/C01MFR03D5W/p1621510734082900)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200368172141330) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by etsybaev**
## What
It's not clear the exact fail reason for https://github.com/airbytehq/airbyte/issues/3549 and jobs IDs were filled without location and project's ID. So presumably at some moment, we may try to update it by the wrong location and get 404.

## How
Added more logs and made the JobId filled with location and projectID. 
Performed a smoke test: migrated some data from Postres to BigQuery sandbox
Selection_277
Selection_279

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [x] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java [hunk 1]
```diff
@@ -96,10 +98,13 @@ public void acceptTracked(AirbyteMessage message) {
             String.format("Message contained record from a stream that was not in the catalog. \ncatalog: %s , \nmessage: %s",
                 Jsons.serialize(catalog), Jsons.serialize(recordMessage)));
       }
+      final BigQueryWriteConfig writer = writeConfigs.get(pair);
       try {
-        final BigQueryWriteConfig writer = writeConfigs.get(pair);
         writer.getWriter().write(ByteBuffer.wrap((Jsons.serialize(formatRecord(writer.getSchema(), recordMessage)) + "\n").getBytes(Charsets.UTF_8)));
-      } catch (IOException e) {
+      } catch (IOException | RetryHelperException e) {
+        LOGGER.error("Got an error while writing message:" + e.getMessage());
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java:105]: let's log as much as we possibly can here if the status is 404: 

* region
* number of input streams 
* all job information 
* any other info that would be helpful

Comment (etsybaev) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java:106]: Hi @sherifnada.
Many thanks for your comment.
I've updated the PR with adding more logs. But the point here is that at the moment when we get an exception - we get it all wrapped as a String. So the only way to check if 404 is "string.contains()" method that looks a little bit ugly for this case as for me. But here is the thing, if we fail at some moment for other reason we anyway would like to know all the details and the whole "sync attempt" would be marked as failed. So I added full logs without additional verification for reason. Please let me know if you don't agree. 
By the way, this PR is in "draft" status for now and waiting for GL internal code review.

File: docs/integrations/destinations/bigquery.md [hunk 2]
```diff
@@ -114,6 +114,7 @@ Therefore, Airbyte BigQuery destination will convert any invalid characters into
 | :--- | :---  | :--- | :--- |
 | 0.3.6 | 2021-06-18 | [#3947](https://github.com/airbytehq/airbyte/issues/3947) | Service account credentials are now optional. |
 | 0.3.4 | 2021-06-07 | [#3277](https://github.com/airbytehq/airbyte/issues/3277) | Add dataset location option |
+| 0.3.9 | 2021-07-28 | [#3549](https://github.com/airbytehq/airbyte/issues/3549) | Add extended logs and made JobId filled with region and projectId |
```
Comment (sherifnada) [inline docs/integrations/destinations/bigquery.md:117]: this should be in reverse chrono order

**Review by sherifnada — APPROVED** [review]

File: docs/integrations/destinations/bigquery.md (hunk 2 above)
Comment (etsybaev) [reply docs/integrations/destinations/bigquery.md]: Hmmm... how didn't I notice it. It was a hard day. Updated, sorry for that

**Review by yaroslav-hrytsaienko — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
shipit

**Pull request merged**